pybamm/expression_tree/symbol.py

```diff
@@ -983,6 +983,18 @@ def shape_for_testing(self):
         else:
             return evaluated_self.shape
 
+    def contains(self, expr_type):
+        """
+        Does the symbol tree contain a given node type?
+        """
+        if type(self) is expr_type:
+            return True
+        elif "children" in dir(self):
+            for child in self.children:
+                if child.contains(expr_type):
+                    return True
+        return False
+
```

Comment:
how is this different from `has_symbol_of_classes` ?

Reply:
@tinosulzer  I hadn't spotted that method - I've reverted to using `has_symbol_of_classes` as the more pythonic implementation to avoid duplicating this behaviour.